backend/app/dependencies.py

```
@@ -0,0 +1,97 @@
+    payload = auth_service.verify_token(token)
+    if payload is None:
+        raise credentials_exception
+    
+    user_id_str = payload.get("sub")
+    if user_id_str is None:
+        raise credentials_exception
+    
+    try:
+        user_id = int(user_id_str)
+    except (ValueError, TypeError):
+        raise credentials_exception
```

Ruff (0.14.8)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)